packages/runtime/src/helpers/functions.ts

```diff
@@ -208,6 +225,143 @@ export const setupImageFunction = async ({
   }
 }
 
+const traceRequiredServerFiles = async (publish: string): Promise<string[]> => {
+  const requiredServerFilesPath = join(publish, 'required-server-files.json')
+  const {
+    files,
+    relativeAppDir,
+    config: {
+      experimental: { outputFileTracingRoot },
+    },
+  } = (await readJSON(requiredServerFilesPath)) as {
+    files: string[]
+    relativeAppDir: string
+    config: { experimental: { outputFileTracingRoot: string } }
+  }
+  const appDirRoot = join(outputFileTracingRoot, relativeAppDir)
+  const absoluteFiles = files.map((file) => join(appDirRoot, file))
+
+  absoluteFiles.push(requiredServerFilesPath)
+
+  return absoluteFiles
+}
+
+const traceNextServer = async (publish: string): Promise<string[]> => {
+  const nextServerDeps = await getDependenciesOfFile(join(publish, 'next-server.js'))
+
+  // during testing, i've seen `next-server` contain only one line.
+  // this is a sanity check to make sure we're getting all the deps.
+  if (nextServerDeps.length < 10) {
+    console.error(nextServerDeps)
+    throw new Error("next-server.js.nft.json didn't contain all dependencies.")
+  }
+
+  const filtered = nextServerDeps.filter((f) => {
```

Comment:
This filter list could be its own array of strings in a const and then we could add more stuff if necessary more easily?

Reply:
I'm fine with replacing this with an array of string, although I don't see a problem with the way it is right now. I could imagine that in the future, there's different kinds of patterns than `.endsWith` that we want to match against, so having this as a filter function is the most versatile.